### PR TITLE
feat: Add `PodPriorityClassName` to `SubmitOpts`. Fixes #7059

### LINKED ISSUE #7059: Possibility to add PriorityClass to workflow submits via REST
# Summary

`podPriorityClassName` is not part of the [submitOpts](https://github.com/argoproj/argo-workflows/blob/4734cbc44dedeb2c7e5984aab5dc9b0c846ff491/pkg/apis/workflow/v1alpha1/common.go#L43). So an API submit does ignore any `PriorityClassName` we set.

# Use Cases

When submitting workflows via REST it is useful to set different `PriorityClasses` depending on timeliness or importance of our workflows.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by nmandery**
Fixes #7059. 

I tested these changes using the development setup described in https://github.com/argoproj/argo-workflows/blob/master/docs/running-locally.md.

Signed-off-by: Nico Mandery <nico@nandery.net>

Don't bother creating a PR until you've done this:

* [x] Run `make pre-commit -B` to fix codegen, lint, and commit message problems.

Create your PR as a draft.

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it
  does not need to pass.
* Once required tests have passed, you can make it "Ready for review".
* Say how how you tested your changes. If you changed the UI, attach screenshots.

Tips:

* If changes were requested, and you've made them, then dismiss the review to get it looked at again.
* Add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md) if you like.
* You can ask for help!

### CODE REVIEW

**Comment by nmandery** [comment]
According to the PR-template PRs should stay as draft until all checks pass. As me being a first-time contributor, CI-checks are disabled for this PR. Could one maintainer please approve the workflows here? Thanks a lot ;)

**Comment by nmandery** [comment]
Thanks for enabling the CI 👍

File: workflow/util/util.go [hunk 1]
```diff
@@ -236,6 +236,10 @@ func ApplySubmitOpts(wf *wfv1.Workflow, opts *wfv1.SubmitOpts) error {
 	if opts.ServiceAccount != "" {
 		wf.Spec.ServiceAccountName = opts.ServiceAccount
 	}
+	if opts.PodPriorityClassName != "" {
```
Comment (alexec) [inline workflow/util/util.go:239]: this needs a test

**Review by alexec — CHANGES_REQUESTED** [review]
needs a test

**Pull request merged**